cpp/harvest/nacjd_data_pub_pipeline.sh

```diff
@@ -7,19 +7,104 @@ if [ $# != 0 ]; then
+    echo "Removing unwanted information (error message) in the file $FILE_NAME"
+    sed -i '/doctype html/d' $FILE_NAME  
+    sed -i '/html lang=en/d' $FILE_NAME 
+    sed -i '/404 Not Found/d' $FILE_NAME 
+    sed -i '/The requested URL was not found/d' $FILE_NAME 
+    sed -i '/Not Found/d' $FILE_NAME 
+    sed -i "s/ //g" $FILE_NAME
+}
 
 
 echo "Extracting ISSN"
 cat "$NACJD_INPUT" | jq -r '.searchResults.response.docs[].ISSN' | sort | uniq > "$ISSN_FILE"
 
 echo "Downloading MARC from K10Plus"
-$ISSN_LOOKUP_K10_PLUS_TOOL "$ISSN_FILE" "$MARC_FILE_K10PLUS"
+$ISSN_LOOKUP_K10_PLUS_TOOL "$ISSN_FILE" "$MARC_FILE_K10PLUS_WITH_DUPS"
+
+echo "Removing duplicatons in $MARC_FILE_K10PLUS_WITH_DUPS"
+marc_remove_dups "$MARC_FILE_K10PLUS_WITH_DUPS" "$MARC_FILE_K10PLUS"
 
 echo "Augmenting MARC using info from K10Plus"
-$NACJD_TOOL "--verbose" $NACJD_INPUT $MARC_FILE_K10PLUS $NACJD_OUTPUT 
\ No newline at end of file
+$NACJD_TOOL "--verbose" "convert" $NACJD_INPUT $MARC_FILE_K10PLUS $ISSN_NOT_FOUND_IN_K10_PLUS $NACJD_OUTPUT
+
+echo "Downloading alternative ISSN from openalex"
+echo "Creating/ Cleaning: $ISSN_ALTERNATIVE_FROM_OPENALEX"
+echo "" > $ISSN_ALTERNATIVE_FROM_OPENALEX
+echo "Starting download data"
+while read ISSN; do
+    echo "$BASE_OPENALEX_ISSN_API$ISSN?select=issn_l,issn"
+    curl -L "$BASE_OPENALEX_ISSN_API$ISSN?select=issn_l,issn" >> $ISSN_ALTERNATIVE_FROM_OPENALEX
+done < $ISSN_NOT_FOUND_IN_K10_PLUS
+
+echo "Removing error in  $ISSN_ALTERNATIVE_FROM_OPENALEX"
+remove_error_message $ISSN_ALTERNATIVE_FROM_OPENALEX
+
+cat $ISSN_ALTERNATIVE_FROM_OPENALEX |jq -r '.issn[]' > $ISSN_ALTERNATIVE_NEED_FROM_K10PLUS
+
+echo "Downloading Alternative MARC from K10Plus"
+$ISSN_LOOKUP_K10_PLUS_TOOL "$ISSN_ALTERNATIVE_NEED_FROM_K10PLUS" "$SOURCE_WITH_DUPS"
+
+echo "Removing duplicatons"
+marc_remove_dups "$SOURCE_WITH_DUPS" "$SOURCE"
+
+echo "Creating $ISSN_ALTERNATIVE_FROM_OPENALEX_CSV"
+cat $ISSN_ALTERNATIVE_FROM_OPENALEX |jq -r '[.issn_l, .issn[]] | flatten | @csv' > $ISSN_ALTERNATIVE_FROM_OPENALEX_CSV
+
+echo "Augmenting 773w"
+$NACJD_TOOL "--verbose" "augment_773w" $NACJD_OUTPUT $ISSN_ALTERNATIVE_FROM_OPENALEX_CSV $SOURCE $NOT_FOUND_ISSN $AUGMENTED_77w_OUTPUT
+
+
+echo "Extracting DOI"
+cat "$NACJD_INPUT" | jq -r '.searchResults.response.docs[].DOI' | sort | uniq > "$NACJD_DOI"
+
+echo "Removing space from $NACJD_DOI"
+sed -i "s/ //g" $NACJD_DOI
+
+echo "Downloading Open Access Info from Openalex"
+echo "Creating/ Cleaning: $JSON_FROM_OPENALEX"
+echo "" > $JSON_FROM_OPENALEX
+echo "Starting download data"
+while read DOI; do
+    echo $DOI
+    curl "$BASE_OPENALEX_DOI_API$DOI?select=doi,open_access" >> $JSON_FROM_OPENALEX
+done < $NACJD_DOI 
+
+echo "Removing error tag from $JSON_FROM_OPENALEX" 
+remove_error_message $JSON_FROM_OPENALEX
+
+echo "Creating open access info CSV: $OPEN_ACCESS_INFO_CSV" 
+cat $JSON_FROM_OPENALEX |jq -r '[.doi, .open_access.is_oa, .open_access.any_repository_has_fulltext] | flatten | @csv' > $OPEN_ACCESS_INFO_CSV
```

Comment:
Is the ISSN licence checking on journal level included if an article has no DOI?